src/syncPersistCache.js

```diff
@@ -0,0 +1,52 @@
+import { CachePersistor } from 'apollo-cache-persist';
+import Persistor from 'apollo-cache-persist/Persistor';
+import Storage from 'apollo-cache-persist/Storage';
+
+export default function(options) {
+  const cachePersistor = new SynchronousCachePersistor(options);
+  cachePersistor.syncRestore();
+}
+
+class SynchronousCachePersistor extends CachePersistor {
+  constructor(options) {
+    super(options);
+
+    this.storage = new SynchronousStorage(options);
+    this.persistor = new SynchronousPersistor(
+      { log: this.log, cache: this.cache, storage: this.storage },
+      options,
+    );
+  }
+
+  syncRestore() {
+    this.persistor.syncRestore();
+  }
+}
+
+class SynchronousPersistor extends Persistor {
+  constructor({ log, cache, storage }, options) {
+    super({ log, cache, storage }, options);
+  }
+
+  syncRestore() {
+    this.cache.restore(this.storage.syncRead());
+  }
+}
+
+class SynchronousStorage extends Storage {
+  constructor(options) {
+    if (
+      options.storage !== window.localStorage &&
```

Comment:
I'm not sure if that is something needed as people can use some wrappers like dixie so having this check might be limiting.

Reply:
Lets not limit this class to only those storage options.

Reply:
My intent was to prevent confusing errors by catching an incompatible storage provider before a write attempt occur. But, I'm fine with removing it.

Reply:
I think we can document it. Problem here is the interface. After merging that change I woul probably come with 2 new storage interfaces one async and one sync. This will  make things more clean.

Reply:
Storage provider checks removed.